### PR TITLE
Make external_hostname and tunnel_name optional

### PR DESCRIPTION
**Pull request by brenner-tobias**
# Proposed Changes

- Enable usage without using _external_host_ (for _custom_config_, _additional_hosts_, _catch_all_service_ or _nginx_proxy_manager_
- Make _tunnel_name_ optional with default "homeassistant"

## Related Issues

#190

### CODE REVIEW

**Comment by brenner-tobias** [comment]
I currently cannot test this properly, so please feel free to suggest any documentation changes or code fixes

File: cloudflared/rootfs/etc/s6-overlay/scripts/10-cloudflared-config.sh [hunk 1]
```diff
@@ -256,10 +259,12 @@ createConfig() {
 createDNS() {
```
Comment (brenner-tobias) [inline cloudflared/rootfs/etc/s6-overlay/scripts/10-cloudflared-config.sh:259]: When looking at this method, I figured we might as well rename 'createCustomDNS' to 'createDNS' and use it all the time. Whether it is a custom_config or a config created by the add-on should not matter, right? @elcajon

Comment (elcajon) [reply cloudflared/rootfs/etc/s6-overlay/scripts/10-cloudflared-config.sh:259]: The difference is we use json for automatically generated config and YAML for custom config. We would need to streamline this first or simply combine both functions which wouldn't really reduce complexity.

**Comment by elcajon** [comment]
Technically, the changes you made work fine. 👍🏻

I think we should adjust the documentation to facilitate the configuration of the add-on. 

The documentation has to make it clear that the choice between local and remote managed tunnel must be made first.

- For using the remote managed tunnel `tunnel_token` needs to be set
- For using the local managed tunnel external_hostname and/or additional_hosts need to be set.
- All other parameters can be used in addition for the local managed tunnel.

In this context (maybe as additional PR), I suggest to remove tunnel_name from the default configuration and set it to the default value "homeassistant" in case of an empty value. Thus we can set the parameter optionally.

File: cloudflared/rootfs/etc/s6-overlay/scripts/10-cloudflared-config.sh (hunk 1 above)
Comment (brenner-tobias) [reply cloudflared/rootfs/etc/s6-overlay/scripts/10-cloudflared-config.sh:259]: Yeah true, I forgot about that. Let's leave it as it was

**Review by elcajon — APPROVED** [review]

**Comment by elcajon** [comment]
Looks good, I would say lets merge this and add documentation in a separate PR

**Comment by brenner-tobias** [comment]
> Looks good, I would say lets merge this and add documentation in a separate PR

sorry, I missed your comment and had another commit lined up. Can you have another quick look @elcajon?

**Comment by elcajon** [comment]
Awesome, looks good. 👍🏻

**Review by elcajon — APPROVED** [review]

**Comment by elcajon** [comment]
I think we could post a new release after this PR and reference the survey again. In parallel we could already remove the warp functionality, the feedback is relatively low on this configuration option.

**Comment by AlecRust** [comment]
You guys are fast! Thank you for this change 🙏 

Looks like you've already sorted the docs, but when this goes in I'll follow it with a minor docs tidy PR.

**Pull request merged**